src/cc/bcc_elf.c

```diff
@@ -435,8 +438,10 @@ static char *find_debug_via_debuglink(Elf *e, const char *binpath,
 
 DONE:
   free(bindir);
-  if (res && check_crc && !verify_checksum(res, crc))
+  if (res && check_crc && !verify_checksum(res, crc)) {
```

Comment:
Huh, I just looked at Man page for `dirname` and it says
```
These functions may return pointers to statically allocated memory which may be overwritten by subsequent calls. Alternatively, they may return a pointer to some part of path, so that the string referred to by path should not be modified or freed until the pointer returned by the function is no longer required.
```
So maybe the `free(bindir)` part is also not really correct per say?

Reply:
I just checked glibc code, For the path without '/' (e.g., a.out), a static memory inside dirname is returned.
It looks the best way is to declare an array, e.g., `bindir[PATH_MAX]` as the input to the bindir function.

This code will not execute in typical case. It will be executed if a separate debug file is specified, which is probably we won't hit the issue often.

Reply:
Good point! The result of strdup(binpath) should be preserved in a tmp pointer and that pointer freed instead of bindir.